### ISSUE TITLE
[ENHANCEMENT] Add support for gestureResponseDistance for iOS pop gestures

### ISSUE THREAD

**Issue by mrousavy** [wix/react-native-navigation#6547]
### Problem

Currently you can only _enable_ or _disable_ the swipe from left edge to right edge gesture to pop screens on iOS. Some apps have this gesture, but instead of only being able to only swipe from the left most edge of the screen, you can swipe from anywhere, such as the middle of the screen (e.g.: Instagram Chats, Telegram)

The react-navigation library has the property [`gestureResponseDistance`](https://reactnavigation.org/docs/stack-navigator/#gestureresponsedistance), which is exactly that. It would be awesome to have that supported in the react-native-navigation library as-well

### Solution

Unfortunately, as solution to this isn't as simple as in the react-navigation library, since we can't simply use some react-native-gesture-handler tricks. Instead, we have to create a custom `UIGestureRecognizer` and bind it to the `interactivePopGestureRecognizer`. I found [this answer on stackoverflow](https://stackoverflow.com/a/22244990/5281431) which also provides a code solution. 

I'm creating this issue as a discussion point, rather than an actual issue. @wix: feel free to close if this isn't going anywhere.

* In case the original iOS `interactivePopGestureRecognizer` changes at some point in the future, we have to mimic that new behavior and adjust it in our copy
* This is a lot of code just for a simple property
    - More to maintain
    - More error prone
    - Behavior isn't really requested by many people

Let me know if you still want me to open a PR with the proposed changes!

**Comment by yogevbd** [comment]
@mrousavy We can easily implement [this solution](https://stackoverflow.com/a/60598558/3562020) and add  `popGestureMode?: 'bezel' | 'entireScreen'` to control it. wdyt?

**Comment by mrousavy** [comment]
@yogevbd That's a good idea, what do you think about just giving the user full control via an `x` coordinate value?

So:

```
popGestureX: Dimensions.get('window').width * 0.1
popGestureX: Dimensions.get('window').width * 0.5
popGestureX: Dimensions.get('window').width
```

**Comment by yogevbd** [comment]
@mrousavy it does add flexibility but I'm not sure it's needed here.. I think that keeping it simple better serves our users

**Comment by mrousavy** [comment]
Makes sense to me. I think I have some time tomorrow/saturday if you're too busy to look into this yourself! @yogevbd

**Comment by hukpavlo** [comment]
@mrousavy @yogevbd Hi, is there an update here? Would be cool to have this feature.

**Comment by mrousavy** [comment]
I managed to implement the full-screen pop gesture:

https://user-images.githubusercontent.com/15199031/122741658-73c97d00-d285-11eb-9bb7-fdedf6488ea9.MP4

but it takes over all other gestures - i.e. I can't even scroll on the chat page anymore. I tried to override the delegate stubs:

```objc
- (BOOL) gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer shouldRecognizeSimultaneouslyWithGestureRecognizer:(UIGestureRecognizer *)otherGestureRecognizer {
  NSLog(@"shouldRecognizeSimultaneouslyWithGestureRecognizer?");
  return NO;
}

- (BOOL) gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer shouldRequireFailureOfGestureRecognizer:(UIGestureRecognizer *)otherGestureRecognizer {
  NSLog(@"shouldRequireFailureOfGestureRecognizer?");
  return YES;
}

- (BOOL) gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer shouldBeRequiredToFailByGestureRecognizer:(UIGestureRecognizer *)otherGestureRecognizer {
  NSLog(@"shouldBeRequiredToFailByGestureRecognizer?");
  return YES;
}
```

but the bottom ones are not called at all, only `shouldRecognizeSimultaneouslyWithGestureRecognizer` is called. Also it makes no difference if I return `YES` or `NO` in that stub.. 🤨

**Comment by ghiculescualexandru** [comment]
Any updates on this?

**Comment by techno-mad** [comment]
@mrousavy any chance of revisiting this? This would be such a "breakthrough" feature, as seemingly every well produced mainstream app allows swiping back from anywhere on the screen. It's to the point where, if an app doesn't allow it - you kind of almost feels like some sort of inferior UX.

**Comment by bogoslavskiy** [comment]
You can see this solution https://github.com/sigmela/react-native-navigation/pull/3

**Comment by yedidyak** [comment]
Closing this as part of a maintainer cleanup of open issues older than one year.

Decision: Closed: non-critical feature request/question/discussion.

If this still reproduces on the latest React Native Navigation with a minimal reproduction, please open a new issue with current RN/RNN versions and repro steps.
